frontend/src/pages/cert-manager/PoliciesPage/components/CertificateProfilesTab/CreateProfileModal.tsx

```diff
@@ -139,10 +152,22 @@ const editSchema = z
       if (data.enrollmentType === "acme" && !data.acmeConfig) {
         return false;
       }
+
+      if (data.issuerType === "ca" && !data.certificateAuthorityId) {
+        return false;
+      }
+      if (data.issuerType === "self-signed" && data.certificateAuthorityId) {
+        return false;
+      }
+      if (data.issuerType === "self-signed" && data.enrollmentType !== "api") {
+        return false;
+      }
+
       return true;
     },
     {
-      message: "Configuration is required for selected enrollment type"
+      message:
+        "Configuration is required for selected enrollment type and issuer type. CA issuer requires a certificate authority. Self-signed issuer cannot have a certificate authority and only supports API enrollment."
     }
```

Comment:
same remark here... I think it's a bit confusing for a user if the message says a lot of different things..